configure.ac

```diff
@@ -5998,6 +6001,48 @@ then
     AM_CFLAGS="$AM_CFLAGS -DHAVE_DH_DEFAULT_PARAMS"
 fi
 
+
+AC_ARG_WITH([max-rsa-bits],
+    [AS_HELP_STRING([--with-max-rsa-bits=number],[number of bits to support for RSA, DH, and DSA keys])],
+    [WITH_MAX_ASSYM_KEY_BITS=$withval],
+    [WITH_MAX_ASSYM_KEY_BITS="$DEFAULT_MAX_ASSYM_KEY_BITS"])
+
+if test -n "$WITH_MAX_ASSYM_KEY_BITS"; then
+   if test "$WITH_MAX_ASSYM_KEY_BITS" -lt 1024 -o "$WITH_MAX_ASSYM_KEY_BITS" -gt 16384; then
+       AC_MSG_ERROR([--with-max-rsa-bits argument must be between 1024 and 16384 inclusive])
+   fi
+   AM_CFLAGS="$AM_CFLAGS -DRSA_MAX_SIZE=$WITH_MAX_ASSYM_KEY_BITS -DFP_MAX_BITS=$((WITH_MAX_ASSYM_KEY_BITS * 2)) -DSP_INT_BITS=$WITH_MAX_ASSYM_KEY_BITS"
+fi
+
+AC_ARG_WITH([max-ecc-bits],
+    [AS_HELP_STRING([--with-max-ecc-bits=number],[number of bits to support for ECC algorithms])],
+    [WITH_MAX_ECC_BITS=$withval],
+    [WITH_MAX_ECC_BITS="$DEFAULT_MAX_ECC_BITS"])
```

Comment:
I don't see `DEFAULT_MAX_ECC_BITS` defined anywhere?

Reply:
uh, yeah, umm, I originally planned to initialize `DEFAULT_MAX_ASSYM_KEY_BITS` and `DEFAULT_MAX_ECC_BITS` with reasonable values near the top of configure.ac, but concluded it was much more important to leave the behavior unchanged and not add a new place in the codebase with a new magical number to remember to coordinate...

we can harmlessly leave `DEFAULT_MAX_ECC_BITS` in that clause and use it in the future should the need arise, or I can nix it.